keep-ui/app/(keep)/topology/ui/map/service-node.tsx

```diff
@@ -73,34 +72,52 @@ function ServiceDetailsTooltip({ data }: { data: TopologyService }) {
 }
 
 export function ServiceNode({ data, selected }: NodeProps<ServiceNodeType>) {
-  const { useAllAlerts } = useAlerts();
-  const { data: alerts, mutate } = useAllAlerts("feed");
-  const { data: pollAlerts } = useAlertPolling();
+  const { data: incidents, mutate: incidentsMutate } = useIncidents(
+    true,
+    25,
+    0,
+    {
+      id: "creation_time",
+      desc: false,
+    },
+    { affected_services: [data.display_name] }
+  );
+  usePollIncidents(incidentsMutate);
   const router = useRouter();
   const [showDetails, setShowDetails] = useState(false);
+  const [tooltipDirection, setTooltipDirection] = useState<Position>(
+    Position.Bottom
+  );
+  const nodeRef = useRef<HTMLDivElement | null>(null);
 
   useEffect(() => {
-    if (pollAlerts) {
-      mutate();
-    }
-  }, [pollAlerts, mutate]);
+    if (!showDetails || !nodeRef.current) return;
 
-  const relevantAlerts = alerts?.filter(
-    (alert) => alert.service === data.service
-  );
+    const rect = nodeRef.current.getBoundingClientRect();
+    const viewportHeight = window.innerHeight;
+    console.log(rect.bottom, viewportHeight, rect);
+
+    if (rect.bottom + 150 > viewportHeight) {
+      setTooltipDirection(Position.Top);
+    } else {
+      setTooltipDirection(Position.Bottom);
+    }
+  }, [showDetails]);
```

Comment:
I'm not sure this is good enough

Uploading CleanShot 2024-12-02 at 12.12.31.mp4…

Reply:
https://github.com/user-attachments/assets/659fc0ab-b059-4f98-ba3a-3860bf2d2964

Reply:
Fixed